internal/internal_workflow.go

```diff
@@ -584,6 +585,17 @@ func (c *channelImpl) receiveAsyncImpl(callback *receiveCallback) (v interface{}
 		r := c.buffer[0]
 		c.buffer[0] = nil
 		c.buffer = c.buffer[1:]
+
+		// Move blocked sends into buffer
+		if len(c.blockedSends) > 0 {
+			b := c.blockedSends[0]
+			c.blockedSends[0] = nil
```

Comment:
Does this line do anything given the next line?

Reply:
No actually. I am just following existing pattern in the code. See Line 606.

Reply:
I believe it would retain memory without this assignment.